pyqir-generator/src/qir/instructions.rs

```diff
@@ -126,5 +126,45 @@ pub(crate) fn emit<'ctx>(
         Instruction::Z(inst) => {
             generator.emit_void_call(generator.qis_z_body(), &[find_qubit(&inst.qubit).into()]);
         }
+        Instruction::If(if_inst) => emit_if(generator, registers, qubits, entry_point, if_inst),
     }
 }
+
+fn emit_if<'ctx>(
+    generator: &CodeGenerator<'ctx>,
+    registers: &mut HashMap<String, Option<PointerValue<'ctx>>>,
+    qubits: &HashMap<String, BasicValueEnum<'ctx>>,
+    entry_point: FunctionValue,
+    if_inst: &If,
+) {
+    // panic if we reference a register that hasn't been declared
+    // default the register value to zero if if hasn't been measured/declared
```

Comment:
This comment may be out of date, in that we do panic if the register hasn't been declared but default to zero if it hasn't been measure (remove the "/declared").